crates/client/settlement/src/errors.rs

```diff
@@ -0,0 +1,30 @@
+use std::time::Duration;
+
+use async_trait::async_trait;
+use futures_timer::Delay;
+use thiserror::Error;
+
+use crate::RetryStrategy;
+
+#[derive(Debug, Error)]
+pub enum UnrecoverableError {
+    #[error("State root mismatch for level {0}")]
+    StateRootMismatch(u64),
+}
+
+pub struct RetryOnRecoverableErrors {
+    pub delay: Duration,
+}
+
+#[async_trait]
+impl RetryStrategy for RetryOnRecoverableErrors {
+    async fn can_retry(&self, error: &anyhow::Error) -> bool {
+        match error.downcast_ref::<UnrecoverableError>() {
+            Some(_) => false,
+            None => {
+                Delay::new(self.delay).await;
+                true
+            }
+        }
+    }
+}
```

Comment:
Where are the recoverable errors defined?

Reply:
Now it's more explicitly defined